gpkit/constraints/set.py

```diff
@@ -127,19 +128,17 @@ def subinplace(self, subs, value=None):
         for constraint in self:
             constraint.subinplace(subs, value)
 
-    @property
-    def varkeys(self):
-        "return all Varkeys present in this ConstraintSet"
-        return self._varkeys()
-
-    def _varkeys(self, init_dict=None):
-        "return all Varkeys present in this ConstraintSet"
-        init_dict = {} if init_dict is None else init_dict
-        out = KeySet(init_dict)
+    def reset_varkeys(self, init_dict=None):
+        "Goes through constraints and collects their varkeys."
+        varkeys = KeySet()
+        if not init_dict is None:
+            varkeys.update(init_dict)
         for constraint in self:
             if hasattr(constraint, "varkeys"):
-                out.update(constraint.varkeys)
-        return out
+                varkeys.update(constraint.varkeys)
+            # else:
+            #     print type(constraint) # getting numpy bools...
+        self.varkeys = varkeys
```

Comment:
I have an external model that now fails on this branch -- the failure occurs on this line by raising an `AttributeError: can't set attribute`. Here's a relevant portion of the traceback (note `lc` here is a LinkedConstraintSet):

```
--> 375         Model.__init__(self, None, lc, **kwargs)
    376
    377     @property

/Users/whoburg/MIT/dev/gpkit/gpkit/constraints/model.pyc in __init__(self, cost, constraints, substitutions, name)
     56         cost = cost if cost else Monomial(1)
     57         constraints = constraints if constraints else []
---> 58         CostedConstraintSet.__init__(self, cost, constraints, substitutions)
     59         if self.__class__.__name__ != "Model" and not name:
     60             name = self.__class__.__name__

/Users/whoburg/MIT/dev/gpkit/gpkit/constraints/costed.pyc in __init__(self, cost, constraints, substitutions)
     18         if substitutions:
     19             subs.update(substitutions)
---> 20         ConstraintSet.__init__(self, constraints, subs)
     21
     22     def subinplace(self, subs, value=None):

/Users/whoburg/MIT/dev/gpkit/gpkit/constraints/set.pyc in __init__(self, constraints, substitutions, recursesubs)
     29             subs.update(constraints.substitutions)  # pylint: disable=no-member
     30         if recursesubs:
---> 31             self.reset_varkeys()
     32             self.substitutions = KeyDict.with_keys(self.varkeys,
     33                                                    self._iter_subs(subs))

/Users/whoburg/MIT/dev/gpkit/gpkit/constraints/costed.pyc in reset_varkeys(self, init_dict)
     30         if init_dict is not None:
     31             costkeys.update(init_dict)
---> 32         ConstraintSet.reset_varkeys(self, costkeys)
     33
     34     def rootconstr_str(self, excluded=None):

/Users/whoburg/MIT/dev/gpkit/gpkit/constraints/set.pyc in reset_varkeys(self, init_dict)
    139             # else:
    140             #     print type(constraint) # getting numpy bools...
--> 141         self.varkeys = varkeys
    142
    143     def as_posyslt1(self):

AttributeError: can't set attribute
```